pandas/core/algorithms.py

```diff
@@ -873,6 +873,9 @@ def value_counts(
             result.name = name
             result.index.name = index_name
             counts = result._values
+            if not isinstance(counts, np.ndarray):
+                # e.g. ArrowExtensionArray
+                counts = np.asarray(counts)
```

Comment:
why is this needed?  id expect pd.value_counts to behave like obj.value_counts

Reply:
For `normalize` below, `counts.sum()` is called and  `._values` which is an ArrowExtensionArray doesn't have `sum`.

I think it was assumed that `counts` is a numpy array here?

Reply:
If EA's implement `value_counts` they should also handle `normalize`?

Reply:
> I think it was assumed that counts is a numpy array here?

For a long time I wanted value_counts to have int64 dtype and get rid of Int64 cases, but eventually gave up on that.  So i think at least Int64 is possible.  IIRC value_counts is only kinda-sorta part of the EA interface so the exact requirements aren't pinned down.

> If EA's implement value_counts they should also handle normalize?

If the normalization logic can live exclusively here, I think I'd prefer that.

> which is an ArrowExtensionArray doesn't have sum.

Out of scope, but I'd just put sum and the other most-basic reductions (say, all the ones in the np.ndarray namespace) on EA and let it raise TypeError for non-supported dtypes.

Reply:
Would you prefer me to put `sum` on ArrowExtensionArray so this works without the `if` statement or should we leave in this check?

Reply:
i guess cleaner to leave this and can do another api change later if called for